pkg/chartutil/create.go

```diff
@@ -112,50 +120,61 @@ const defaultIgnore = `# Patterns to ignore when building packages.
 `
 
 const defaultIngress = `{{- if .Values.ingress.enabled -}}
-{{- $serviceName := include "<CHARTNAME>.fullname" . -}}
-{{- $servicePort := .Values.service.externalPort -}}
+{{- $fullName := include "<CHARTNAME>.fullname" . -}}
+{{- $servicePort := .Values.service.port -}}
+{{- $ingressPath := .Values.ingress.path -}}
 apiVersion: extensions/v1beta1
 kind: Ingress
 metadata:
-  name: {{ template "<CHARTNAME>.fullname" . }}
+  name: {{ $fullName }}
   labels:
     app: {{ template "<CHARTNAME>.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    chart: {{ template "<CHARTNAME>.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- with .Values.ingress.annotations }}
   annotations:
-    {{- range $key, $value := .Values.ingress.annotations }}
-      {{ $key }}: {{ $value | quote }}
-    {{- end }}
+{{ toYaml . | indent 4 }}
+{{- end }}
 spec:
+{{- if .Values.ingress.tls }}
+  tls:
+  {{- range .Values.ingress.tls }}
+    - hosts:
+      {{- range .hosts }}
+        - {{ . }}
+      {{- end }}
+      secretName: {{ .secretName }}
+  {{- end }}
+{{- end }}
   rules:
-    {{- range $host := .Values.ingress.hosts }}
-    - host: {{ $host }}
+  {{- range .Values.ingress.hosts }}
+    - host: {{ . }}
       http:
         paths:
-          - path: /
+          - path: {{ $ingressPath }}
             backend:
-              serviceName: {{ $serviceName }}
-              servicePort: {{ $servicePort }}
-    {{- end -}}
-  {{- if .Values.ingress.tls }}
-  tls:
-{{ toYaml .Values.ingress.tls | indent 4 }}
-  {{- end -}}
-{{- end -}}
+              serviceName: {{ $fullName }}
+              servicePort: http
+  {{- end }}
+{{- end }}
 `
 
-const defaultDeployment = `apiVersion: extensions/v1beta1
+const defaultDeployment = `apiVersion: apps/v1beta2
```

Comment:
we could use start using `apps/v1`

Reply:
We officially support the current and previous minor versions of Kubernetes over at the charts repo which currently would be 1.8+. `apps/v1` requires 1.9.

Reply:
IMHO we should always default to the latest stable version (atleast for new charts), otherwise its just another `todo` once `v1beta2` is officially deprecated. 

Users can always change this back to `v1beta2` if they have a compelling reason. 

will it break the CI?

Reply:
CI is at 1.8, so it will break. And it will cause extra work in code reviews when we have to tell people to change it.

Reply:
Agreed.

Reply:
Tectonic is always an entire release behind.
Is 1.9 considered the stable release?

Reply:
Kubernetes itself, as of today, supports 1.9, 1.8, and 1.7. The charts repo, for the moment, supports 1.9 and 1.8. Eventually I would like to see support for 3 minor versions. GKE and many other services don't have production 1.9 out yet, even though it's been released for over a month.

`apps/v1` is only supported in 1.9. Many people are not on 1.9, yet.

I would suggest updating to `apps/v1` when 1.10 is out, at the earliest.

Reply:
You can add Tectonic to that list too. It released 1.8.4 in December.